src/main/scala/Hash.scala

```diff
@@ -17,16 +17,16 @@ final class Hash(size: Int) {
     roleIndex(piece.role) * 2 + piece.color.fold(1, 0)
 
   private def posIndex(pos: Pos) =
-    8 * (pos.y - 1) + (pos.x - 1)
+    8 * pos.y + (pos.x - 9)
```

Comment:
Not measurable speed-up in these lines, but just less arithmetic operations.
All speed-ups achieved by replacing vector xor operations by xor operations on primitive long type. No column transpose, no index range checks, no conversion toByte and 3 times less xor operation even in case of small PositionHash.